NitroxClient/MonoBehaviours/Gui/MainMenu/JoinServer.cs

```diff
@@ -158,6 +161,16 @@ private void StartMultiplayerClient()
             }
             catch (ClientConnectionFailedException)
             {
+                multiplayerSession.Client.Stop();
```

Comment:
Not sure if this works. Seems very hacky to just jump networking implementation on error in the hope it'll work. I think it's better to just let the client crash/close and restart using LiteNetLibClient.

Reply:
While testing this works, but I will have a look tomorrow for a better solution.

Reply:
Oof - I feel like this misses the mark. A simple state pattern hidden behind the IClient interface is really all that is needed to handle this problem. You could have a state for Lidgren and a state for LitNetLib. The default state would be whatever the user selected, and the states could "fallback" to each other in the event one of them fails.

Hit me up in Discord if you'd like some pointers on this and I'd be happy to work you with. A few simple refactors and you'll be off to the races!